torch/fft/__init__.py

```diff
@@ -87,6 +87,98 @@
     tensor([0.+0.j, 1.+0.j, 2.+0.j, 3.+0.j])
 """)
 
+fftn = _add_docstr(_fft.fft_fftn, r"""
+fftn(input, s=None, dim=None, norm=None) -> Tensor
+
+Computes the N dimensional discrete Fourier transform of :attr:`input`.
+
+Note:
+
+    The Fourier domain representation of any real signal satisfies the
+    Hermitian property: ``X[i_1, ..., i_n] = conj(X[-i_1, ..., -i_n])``. This
+    function always returns all positive and negative frequency terms, even
+    though for real inputs, half of these values are redundant.
+    :func:`~torch.fft.rfftn` returns the more efficient "half-complex"
+    representation where only the positive frequencies of the last dimension
+    are stored explicitly.
+
+Args:
+    input (Tensor): the input tensor
+    s (Tuple[int], optional): Signal size in the transformed dimensions. If
+        given, the input will either be zero-padded or trimmed to this length
+        before computing the FFT.
+    dim (Tuple[int], optional): Dimensions to be Fourier transformed.
```

Comment:
"Dimensions to be transformed" is sufficient. If s and dim are given, do they have to be the same size?

In the NumPy version of this function repeated dims are allowed, although the NumPy documentation isn't very clear about what happens when they're provided. What happens if dims are repeated here?

Reply:
>In the NumPy version of this function repeated dims are allowed, although the NumPy documentation isn't very clear about what happens when they're provided. What happens if dims are repeated here?

In NumPy, if you give the same dimension multiple times it will FFT that dimension twice which I don't think is a meaningful transformation. This PR follows the SciPy interface which checks for unique dimensions. That was done because we though it was more likely to be user error than a desirable feature.

Supporting repeated transforms would be possible but would require more complication to support. It also wouldn't be any much efficient than just calling `fftn` multiple times.